core/chains/evm/chain.go

```diff
@@ -73,18 +134,52 @@ func (e errChainDisabled) Error() string {
 	return fmt.Sprintf("cannot create new chain with ID %s, the chain is disabled", e.ChainID.String())
 }
 
-func newTOMLChain(ctx context.Context, chain *toml.EVMConfig, opts ChainSetOpts) (*chain, error) {
+// TODO BCF-2509 what is this and does it need the entire app config?
+type GeneralConfig interface {
+	config.AppConfig
+	toml.HasEVMConfigs
+}
+
+type ChainRelayExtenderConfig struct {
+	Logger   logger.Logger
+	DB       *sqlx.DB
+	KeyStore keystore.Eth
+	RelayerConfig
+}
+
+// options for the relayer factory.
+// TODO BCF-2508 clean up configuration of chain and relayer after BCF-2440
+// the factory wants to own the logger and db
+// the factory creates extenders, which need the same and more opts
+type RelayerConfig struct {
+	GeneralConfig GeneralConfig
+
+	EventBroadcaster   pg.EventBroadcaster
+	MailMon            *utils.MailboxMonitor
+	GasEstimator       gas.EvmFeeEstimator
+	OperationalConfigs evmtypes.Configs
+
+	// Gen-functions are useful for dependency injection by tests
+	GenEthClient      func(*big.Int) client.Client
+	GenLogBroadcaster func(*big.Int) log.Broadcaster
+	GenLogPoller      func(*big.Int) logpoller.LogPoller
+	GenHeadTracker    func(*big.Int, httypes.HeadBroadcaster) httypes.HeadTracker
+	GenTxManager      func(*big.Int) txmgr.TxManager
+	GenGasEstimator   func(*big.Int) gas.EvmFeeEstimator
```

Comment:
Were these already leaking out of tests in to production the exported API?

Reply:
yes, sadly, and they are used heavily in cltest and evmtest